api/integrations/launch_darkly/launch_darkly.py

```diff
@@ -0,0 +1,239 @@
+from features.multivariate.models import (
+    MultivariateFeatureOption,
+    MultivariateFeatureStateValue,
+)
+from features.value_types import BOOLEAN, FLOAT, INTEGER, STRING
+from integrations.launch_darkly.client import LaunchDarklyClient
+from integrations.launch_darkly.models import LaunchDarklyImport
+from integrations.launch_darkly.serializers import LaunchDarklyImportSerializer
+from organisations.models import Organisation
+from projects.models import Project
+from users.models import FFAdminUser
+
+
+class LaunchDarklyWrapper:
+    def __init__(
+        self,
+        api_key: str,
+        user: FFAdminUser,
+        request: LaunchDarklyImportSerializer,
+        client: LaunchDarklyClient = None,
+    ):
+        self.client = client or LaunchDarklyClient(api_key)
+        self.request = request
+        self.logger = LaunchDarklyImport.objects.create(
+            user=user,
+            organisation=self.request["organisation_id"],
+            project=self.request["project_id"],
+        )
+        self.import_id = self.logger.uuid
+
+    def import_data(self):
+        self.logger.info(
+            f"Starting Launch Darkly feature importer with job id: {self.import_id}"
+        )
+        organisation_id = self.request["organisation_id"]
+        self.logger.info(
+            f"Attempting to retrieve Flagsmith organisation with id: {organisation_id}"
+        )
+        organisation = Organisation.objects.get(pk=self.request["organisation_id"])
+        self.logger.info(
+            f"Successfully retrieved Flagsmith organisation with id: {organisation_id}"
+        )
+        ld_project_id = self.request["ld_project_id"]
+        try:
+            project_id = self.request["project_id"]
+            self.logger.info(
+                f"Attempting to retrieve Flagsmith project with id: {project_id}"
+            )
+            project = Project.objects.get(project_id)
+            self.logger.info(
+                f"Successfully retrieved Flagsmith project with id: {project_id}"
+            )
+        except KeyError:
+            self.logger.warning(
+                "Could not find Flagsmith project with id: {project_id}"
+            )
+            self.logger.info(
+                "Creating new Flagsmith project for Launch Darklys' imported features"
+            )
+            ld_project = self.client.get_project(ld_project_id)
+            project = self._create_project(organisation, ld_project)
+            self.logger.info(
+                "Successfully created new Flagsmith project for Launch Darklys' imported features"
+            )
+
+        self.logger.info(
+            f"Attempting to retrieve Launch Darkly environments under project id: {ld_project_id}"
+        )
+        ld_environments = self.client.get_environments(ld_project_id)
+        if not ld_environments:
+            # todo do we want to stop the script if no environments can be found
```

Comment:
Hmm... good question - I guess that as long as we can still create the features, there's no issue.